include/kframework/evm.md

```diff
@@ -1385,6 +1435,23 @@ The various `CALL*` (and other inter-contract control flow) operations will be d
 
     rule #computeValidJumpDestsWithinBound(PGM, I, RESULT) => #computeValidJumpDests(PGM, I +Int 1, RESULT ListItem(I)) requires PGM [ I ] ==Int 91
     rule #computeValidJumpDestsWithinBound(PGM, I, RESULT) => #computeValidJumpDests(PGM, I +Int #widthOpCode(PGM [ I ]), RESULT) requires notBool PGM [ I ] ==Int 91
+
+   syntax Map ::= #computeBasicBlocks(Bytes, List)           [function, memo]
+                | #computeBasicBlocksAux(Bytes, List, Int, Map) [function, klabel(#computeBasicBlocksAux)]
+ // ---------------------------------------------------------------------------------------------------
+   rule #computeBasicBlocks(PGM, .List)                 => 0 |-> PGM
+   rule #computeBasicBlocks(PGM, ListItem(0) JUMPDESTS) => #computeBasicBlocksAux(PGM, JUMPDESTS ListItem(lengthBytes(PGM)), 0, .Map)
+   rule #computeBasicBlocks(PGM, JUMPDESTS)             => #computeBasicBlocksAux(PGM, JUMPDESTS ListItem(lengthBytes(PGM)), 0, .Map) [owise]
+
+   rule #computeBasicBlocksAux(_PGM, .List, _, RESULT) => RESULT
+   rule #computeBasicBlocksAux(PGM,
+                            ListItem(NEXT_BB_START) JUMPDESTS,
+                            CURRENT_BB_START,
+                            RESULT) =>
+        #computeBasicBlocksAux(PGM,
+                            JUMPDESTS,
+                            NEXT_BB_START,
+                            RESULT (CURRENT_BB_START |-> #range(PGM, CURRENT_BB_START, NEXT_BB_START -Int CURRENT_BB_START)))
```

Comment:
Hmmm, it seems to me that relying on `qsort` makes this at least `n log(n)`, whereas just iterating over the original bytecode makes it `n`? Basically just consume the bytecode 1 byte at a time, and look for exactly the byte that is `JUMPDEST`, hwile keeping track of the offset, and then emit the map as you go?

Reply:
Yeah, makes sense, it's indeed easier to just traverse the bytecode the second time than trying to reuse the computed jumpdests. Done.